client/services/SolutionHandlerService.js

```diff
@@ -60,7 +60,9 @@ tie.factory('SolutionHandlerService', [
               task, codeEvalResult, codeSubmission.getRawCodeLineIndexes());
             TranscriptService.recordSnapshot(
               SnapshotObjectFactory.create(codeEvalResult, runtimeFeedback));
-            return runtimeFeedback;
+            var finalFeedback = ReinforcementGeneratorService.getReinforcement(
```

Comment:
Don't modify the feedback object. Instead, when passing stuff to the learner view directive, pass the reinforcement object should be passed as a separate object from the usual feedback. Do all the computations in services so that there is no logic in the directive controller.